tomviz/ViewMenuManager.cxx

```diff
@@ -199,4 +226,126 @@ void ViewMenuManager::setShowOrientationAxes(bool show)
   }
 }
 
+void ViewMenuManager::updateDataSource(DataSource* s)
+{
+  m_dataSource = s;
+  updateDataSourceEnableStates();
+}
+
+void ViewMenuManager::updateDataSourceEnableStates()
+{
+  m_showDarkDataAction->setEnabled(m_dataSource && m_dataSource->darkData());
+  m_showWhiteDataAction->setEnabled(m_dataSource && m_dataSource->whiteData());
+}
+
+void ViewMenuManager::showDarkData()
```

Comment:
The showDarkData and showWhiteData code should be combined, and just take the appropriate image as input rather than duplicating so much code. It might be better to simply offer a combobox where the user can switch between the two rather than making them click around.